include/pybind11/cast.h

```diff
@@ -374,11 +374,31 @@ class type_caster_generic {
     object temp;
 };
 
-/* Determine suitable casting operator */
+/** Determine suitable casting operator for pointer-or-lvalue-casting type casters.  The type caster
+ * needs to provide `operator T*()` and `operator T&()` operators.
```

Comment:
Yeah, that does look better.  37b2383 does this on master, and I'll update this PR.